src/execution/sql/runtime_types.cpp

```diff
@@ -655,4 +658,711 @@ Timestamp Timestamp::FromYMDHMSMU(int32_t year, int32_t month, int32_t day, int3
+  if (pos == input.size()) {
+    for (uint32_t i = 0; i < precision - 1; i++) {
+      value_ *= 10;
+    }
+    if (is_negative) {
+      value_ = -value_;
+    }
+    return;
+  }
+  // Skip decimal point
+  pos++;
+  // Nothing after decimal point case
+  if (pos == input.size()) {
+    for (uint32_t i = 0; i < precision - 1; i++) {
+      value_ *= 10;
+    }
+    if (is_negative) {
+      value_ = -value_;
+    }
+    return;
+  }
+
+  for (uint32_t i = 1; i < precision; i++) {
+    if (pos < input.size()) {
+      value_ += input[pos] - '0';
+      value_ *= 10;
+      pos++;
+    } else {
+      for (uint32_t j = i; j < precision; j++) {
+        value_ *= 10;
+      }
+      if (is_negative) {
+        value_ = -value_;
+      }
+      return;
+    }
+  }
+
+  if (pos == input.size()) {
+    if (is_negative) {
+      value_ = -value_;
+    }
+    return;
+  }
+
+  if (pos == input.size() - 1) {
+    // No Rounding required
+    value_ += input[pos] - '0';
+  } else {
+    if (input[pos + 1] - '0' > 5) {
+      // Round Up
+      value_ += input[pos] - '0' + 1;
+    } else if (input[pos + 1] - '0' < 5) {
+      // No Rounding will happen
+      value_ += input[pos] - '0';
+    } else {
+      if ((input[pos] - '0') % 2 == 0) {
+        // Round up if ODD
+        value_ += input[pos] - '0';
+      } else {
+        // Round up if ODD
+        value_ += input[pos] - '0' + 1;
+      }
+    }
+  }
+
+  if (is_negative) {
+    value_ = -value_;
+  }
+}
+
+void CalculateMultiWordProduct128(const uint128_t *const half_words_a, const uint128_t *const half_words_b,
+                                  uint128_t *half_words_result, uint32_t m, uint32_t n) {
+  uint128_t k, t;
+  uint32_t i, j;
+  constexpr const uint128_t bottom_mask = (uint128_t{1} << 64) - 1;
+  for (i = 0; i < m; i++) half_words_result[i] = 0;
+  for (j = 0; j < n; j++) {
+    k = 0;
+    for (i = 0; i < m; i++) {
+      t = half_words_a[i] * half_words_b[j] + half_words_result[i + j] + k;
+      half_words_result[i + j] = t & bottom_mask;
+      k = t >> 64;
+    }
+    half_words_result[j + m] = k;
+  }
+}
+
+int Nlz128(uint128_t x) {
+  // Not used Figure 5-19 - Hacker's Delight double method as we need for 128 bits
+  constexpr uint128_t a = (static_cast<uint128_t>(0x0000000000000000) << 64) | 0xFFFFFFFFFFFFFFFF;
+  constexpr uint128_t b = (static_cast<uint128_t>(0x00000000FFFFFFFF) << 64) | 0xFFFFFFFFFFFFFFFF;
+  constexpr uint128_t c = (static_cast<uint128_t>(0x0000FFFFFFFFFFFF) << 64) | 0xFFFFFFFFFFFFFFFF;
+  constexpr uint128_t d = (static_cast<uint128_t>(0x00FFFFFFFFFFFFFF) << 64) | 0xFFFFFFFFFFFFFFFF;
+  constexpr uint128_t e = (static_cast<uint128_t>(0x0FFFFFFFFFFFFFFF) << 64) | 0xFFFFFFFFFFFFFFFF;
+  constexpr uint128_t f = (static_cast<uint128_t>(0x3FFFFFFFFFFFFFFF) << 64) | 0xFFFFFFFFFFFFFFFF;
+  constexpr uint128_t g = (static_cast<uint128_t>(0x7FFFFFFFFFFFFFFF) << 64) | 0xFFFFFFFFFFFFFFFF;
+
+  // clang made me comment this
+  // if (x == 0) return (128);
```

Comment:
I don't understand why. I can check for myself in a bit.

Reply:
because shift by 128 is not allowed